docs/sources/oncall-api-reference/schedules.md

```diff
@@ -197,3 +197,138 @@ curl "{{API_URL}}/api/v1/schedules/SBM7DV7BKFUYU/" \
+      "shift_start": "2023-01-13T09:00:00Z",
+      "shift_end": "2023-01-13T17:00:00Z"
+    },
+    {
+      "user_pk": "UC2CHRT5SD34X",
+      "shift_start": "2023-01-16T09:00:00Z",
+      "shift_end": "2023-01-16T17:00:00Z"
+    },
+    {
+      "user_pk": "U7S8H84ARFTGN",
+      "shift_start": "2023-01-18T09:00:00Z",
+      "shift_end": "2023-01-18T17:00:00Z"
+    },
+    {
+      "user_pk": "UC2CHRT5SD34X",
+      "shift_start": "2023-01-20T09:00:00Z",
+      "shift_end": "2023-01-20T17:00:00Z"
+    },
+    {
+      "user_pk": "U7S8H84ARFTGN",
+      "shift_start": "2023-01-23T09:00:00Z",
+      "shift_end": "2023-01-23T17:00:00Z"
+    },
+    {
+      "user_pk": "UC2CHRT5SD34X",
+      "shift_start": "2023-01-25T09:00:00Z",
+      "shift_end": "2023-01-25T17:00:00Z"
+    },
+    {
+      "user_pk": "U7S8H84ARFTGN",
+      "shift_start": "2023-01-27T09:00:00Z",
+      "shift_end": "2023-01-27T17:00:00Z"
+    }
+  ]
+}
+```
+
+## Caveats
+
+Some notes on the `start_date` and `end_date` query parameters:
+
+- they are both required and should represent ISO 8601 formatted dates
+- `end_date` must be greater than or equal to `start_date`
+- `end_date` cannot be more than 365 days in the future from `start_date`
+
+Lastly, this endpoint is currently only active for web schedules. It will return HTTP 400 for schedules
+defined via Terraform or iCal.
+
+## Example script to transform data to .csv for all of your schedules
+
+The following Python script will generate a `.csv` file, `oncall-report-2023-01-01-to-2023-01-31.csv`. This file will
+contain two columns, `user_pk` and `hours_on_call`, which represents how many hours each user was on call during the
+period starting January 1, 2023 to January 31, 2023 (inclusive).
+
+```python
+import collections
+import csv
+import requests
+from datetime import datetime
+
+# CUSTOMIZE THE FOLLOWING VARIABLES
+START_DATE = "2023-01-01"
+END_DATE = "2023-01-31"
+OUTPUT_FILE_NAME = f"oncall-report-{START_DATE}-to-{END_DATE}.csv"
+MY_ONCALL_API_BASE_URL = "https://oncall-prod-us-central-0.grafana.net/oncall/api/v1/schedules"
+MY_ONCALL_API_KEY = "meowmeowwoofwoof"
+
+headers = {"Authorization": MY_ONCALL_API_KEY}
+schedule_ids = [schedule["id"] for schedule in requests.get(MY_ONCALL_API_BASE_URL, headers=headers).json()["results"]]
+user_on_call_hours = collections.defaultdict(int)
+
+for schedule_id in schedule_ids:
+  response = requests.get(
+    f"{MY_ONCALL_API_BASE_URL}/{schedule_id}/final_shifts?start_date={START_DATE}&end_date={END_DATE}",
+    headers=headers)
+
+  for final_shift in response.json()["results"]:
+    end = datetime.fromisoformat(final_shift["shift_end"])
+    start = datetime.fromisoformat(final_shift["shift_start"])
+    shift_time_in_seconds = (end - start).total_seconds()
+    user_on_call_hours[final_shift["user_pk"]] += shift_time_in_seconds / (60 * 60)
+
+with open(OUTPUT_FILE_NAME, "w") as fp:
+  csv_writer = csv.DictWriter(fp, ["user_pk", "hours_on_call"])
```

Comment:
I guess it will be useful to request user's e-mail here and write it to the file.

Reply:
good idea 👍 implemented in [this commit](https://github.com/grafana/oncall/pull/2047/commits/f9accf5b400d0b9ee636386b72391e23fe263446)